pkg/process/checks/process_nix_test.go

```diff
@@ -427,3 +442,705 @@ func TestProcessGPUTagging(t *testing.T) {
+		expectedStat := &model.ProcessStat{
+			Pid:                    wlmProc.Pid,
+			CreateTime:             stats.CreateTime,
+			Memory:                 formatMemory(stats),
+			Cpu:                    formatCPU(stats, lastStats, procutilCPUTimeToCPUTimeStat(stats.CPUTime), procutilCPUTimeToCPUTimeStat(lastStats.CPUTime)),
+			Nice:                   stats.Nice,
+			Threads:                stats.NumThreads,
+			OpenFdCount:            stats.OpenFdCount,
+			ProcessState:           model.ProcessState(model.ProcessState_value[stats.Status]),
+			IoStat:                 formatIO(stats, lastStats.IOStat, now, lastRunTime),
+			VoluntaryCtxSwitches:   uint64(stats.CtxSwitches.Voluntary),
+			InvoluntaryCtxSwitches: uint64(stats.CtxSwitches.Involuntary),
+			//ContainerId:            wlmProc.ContainerID, // TODO: current not tested, will be done when container data is added
+		}
+		expectedStats[i] = expectedStat
+	}
+	return expectedStats
+}
+
+func TestProcessCheckRunWLM3Times(t *testing.T) {
+	for _, tc := range []struct {
+		name                string
+		elevatedPermissions bool
+		realtimeCollection  bool
+	}{
+		{
+			name:                "realtime DISABLED, elevated permissions DISABLED",
+			elevatedPermissions: false,
+			realtimeCollection:  false,
+		},
+		{
+			name:                "realtime DISABLED, elevated permissions ENABLED",
+			elevatedPermissions: true,
+			realtimeCollection:  false,
+		},
+		{
+			name:                "realtime ENABLED, elevated permissions DISABLED",
+			elevatedPermissions: false,
+			realtimeCollection:  true,
+		},
+		{
+			name:                "realtime ENABLED, elevated permissions ENABLED",
+			elevatedPermissions: true,
+			realtimeCollection:  true,
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+
+			processCheck, mockProbe, mockWLM := processCheckWithMockProbeWLM(t, tc.elevatedPermissions)
+			now := time.Now()
+			constantClock1 := clock.NewMock()
+			processCheck.timer = constantClock1
+
+			// CREATE TEST DATA
+			wlmProcs, statsByPid1 := createTestWLMProcessData(now, tc.elevatedPermissions)
+			proc1, proc2, proc3, proc4, proc5, proc6 := wlmProcs[0], wlmProcs[1], wlmProcs[2], wlmProcs[3], wlmProcs[4], wlmProcs[5]
+
+			// MOCK WLM
+			mockWLM.On("ListProcesses").Return(wlmProcs)
+
+			// MOCK PROBE FOR STATS
+			pids := make([]int32, len(wlmProcs))
+			for i, wlmProc := range wlmProcs {
+				pids[i] = wlmProc.Pid
+			}
+			mockProbe.On("StatsForPIDs", pids, mock.Anything).Return(statsByPid1, nil).Once()
+
+			// TEST FUNCTION
+			actual, err := processCheck.runWLM(0, tc.realtimeCollection)
+
+			// The first run returns nothing because processes must be observed on two consecutive runs
+			expected1 := CombinedRunResult{}
+			require.NoError(t, err)
+			assert.Equal(t, expected1, actual)
+			// creation times should be the same
+			for _, wlmProc := range wlmProcs {
+				assert.Equal(t, wlmProc.CreationTime.Unix(), statsByPid1[wlmProc.Pid].CreateTime)
+			}
```

Comment:
I'm confused about why this needs to be checked. Doesn't seem to be able to change since the return is mocked and static.